mono/metadata/object.c

```diff
@@ -7328,21 +7271,16 @@ mono_string_from_utf32_checked (mono_unichar4 *data, MonoError *error)
 }
 
 static char *
-mono_string_to_utf8_internal (MonoMemPool *mp, MonoImage *image, MonoString *s, gboolean ignore_error, MonoError *error)
+mono_string_to_utf8_internal (MonoMemPool *mp, MonoImage *image, MonoString *s, MonoError *error)
+// In order to share code, error is optional.
 {
-	MONO_REQ_GC_UNSAFE_MODE;
-
 	char *r;
 	char *mp_s;
 	int len;
 
-	if (ignore_error) {
-		r = mono_string_to_utf8_ignore (s);
-	} else {
-		r = mono_string_to_utf8_checked (s, error);
-		if (!mono_error_ok (error))
-			return NULL;
-	}
+	r = mono_string_to_utf8_checked (s, error);
+	if (error && !mono_error_ok (error))
```

Comment:
Same as https://github.com/mono/mono/pull/6268/files#diff-55205fbc5be02187cdfe6d77547ab70fR65 and you should use `return_val_if_nok(error, NULL)`

Reply:
The point was to have less code, fewer functions, more shared code. In particular, the code duplication I had noticed, due to all the checked/ignore forking, is the thing about padding out utf8 conversion with zeros.